code/player/controllers/ChairController.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class ChairController : PlayerController
+{
+    /// <summary>
+    /// Enables spawning cups in the left and right armrests by
+    /// pressing <c>slot3</c> and <c>slot4</c> respectively.
+    /// </summary>
+    [ConVar.Replicated("cinema.chair.debug")]
+    public static bool ChairDebug { get; set; } = false;
+
+    [Net]
+    public CinemaChair Chair { get; set; }
+
+    protected override void OnActivate()
+    {
+        base.OnActivate();
+
+        SinceActivated = 0f;
+    }
+
+    protected TimeSince SinceActivated { get; set; }
+
+    public override void Simulate(IClient cl)
+    {
+        base.Simulate(cl);
+        Entity.Rotation = Chair.Rotation;
+
+        Entity.EyeRotation = Entity.LookInput.ToRotation();
+        Entity.EyeLocalPosition = Entity.Transform.PointToLocal(LookPosition);
+
+        SimulateAnimation();
+
+        if (Game.IsClient)
+        {
+            return;
+        }
+
+        // Wait a while after activing to check whether the use button is pressed,
+        // as the input to begin using the chair may have occurred on the same tick.
+        if (Input.Pressed("use") && SinceActivated > Time.Delta)
+        {
+            Log.Trace($"{Entity.Client} - Stopped sitting in chair: {Chair.Name}");
+            Chair.EjectUser();
+            return;
+        }
+
+        HandleArmrest(Armrest.Sides.Left, "slot1", "slot3");
+        HandleArmrest(Armrest.Sides.Right, "slot2", "slot4");
+    }
+
+    private void HandleArmrest(Armrest.Sides side, string toggleSlot, string debugSlot)
+    {
+        if (Input.Pressed(toggleSlot))
+        {
+            Chair.Armrests[side].Toggle();
+        }
+        else if (ChairDebug && Input.Pressed(debugSlot))
+        {
+            var armrest = Chair.Armrests[side];
+            var armrestIsLowered = armrest.State == Armrest.States.Lowered;
+            if (armrestIsLowered && armrest.Cupholded == null)
+            {
+                PrimeTest(side);
+            }
+        }
+    }
+
+    private void PrimeTest(Armrest.Sides side)
```

Comment:
I'm assuming this will be removed later so i'm ignoring any debug slot / test code for this.